src/frequenz/sdk/actor/power_distributing/battery_pool.py

```diff
@@ -0,0 +1,125 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+"""Class that stores pool of batteries and manage them."""
+
+import asyncio
+import logging
+from asyncio import ALL_COMPLETED
+from typing import Set
+
+from ..._data_ingestion.battery_utils import BatteryUtils, WorkingState
+from .result import PartialFailure, Result, Success
+
+_logger = logging.getLogger(__name__)
+
+
+class BatteryPool:
```

Comment:
Yes, but now it needs PowerDistirbutingResult. And PowerDistributing needs BatteryPool. And that is circular dependency  :( 
I the future it will be moved.
I will make it clear in commit message.

Reply:
> Also, we need to start asking the question, is this class intended to be used by SDK users or it is internal, because if it internal it should be in a private module (starting with _).

It is intended to be used by users in the future

Reply:
OK, as long as you are having what should be public interface and what should be internal and hidden, all good if we need to go through an intermediary step where things are not where they should in a final version :+1:

Reply:
(OTOH as long as stuff is temporary I would still make the module hidden to make it clear for users that if they use that module it can disappear without notice at any time)

Reply:
OK I made it private